src/test/ui/conditional-compilation/cfg_accessible-bugs.rs

```diff
@@ -0,0 +1,24 @@
+// check-fail
+// known-bug
+
+// This should pass.
+
+#![feature(cfg_accessible)]
+
+enum Enum {
+    Existing { existing: u8 },
+}
+
+// FIXME: Currently returns `false` but should be "not sure"
+#[cfg_accessible(Enum::Existing::existing)]
+const A: bool = true;
+
+// FIXME: Currently returns `false` but should be "not sure"
```

Comment:
I have put a hack in https://github.com/rust-lang/rust/pull/97391/commits/a6156ca8588865f1c2b42220fac4411f8dfd2131 that seems to work but it's still a hack.

I'm pretty sure the problem lies [here](https://github.com/rust-lang/rust/blob/9fed13030c2a2ebd79bfb1fd8be4f768cbe8c9d9/compiler/rustc_resolve/src/ident.rs#L1510-L1517), I think it should return an `NonModule` instead of a `failure` like the rest of the conditions. I tried doing this but it's not trivial many diagnostics change sometimes dramatically (the error reported no longer make sense). See https://github.com/rust-lang/rust/commit/396ccf426934433debb0abc600dc43ce29b08214 for more information.

Reply:
The `PathResult` and cases in which its variants are returned are pretty inconsistent.
It would be nice to refactor it to always keep partial resolutions, regardless of module-ness, but it indeed may affect some wider range of diagnostics (https://github.com/rust-lang/rust/commit/2ddd1bc4aeb8f50a8c42e2d1af17e819ebfb843e doesn't look like a correct or complete solution for that).